### PR TITLE
The MVC design pattern set up for server

### LINKED ISSUE #30: Folder Restructure the Server App
## Context
We want our app decoupled and module based so that it is easy to maintain. 

## Problem
Now our server is coded everything in one file (server.js) but I should not be like that.

## Solution 

- [ ] create different folder for different microservices "account_management", "schedule_planner", "mood_tracker" etc.
- [ ] create three sub folder inside each folder "routes", "controllers", "models" and a mini express app similar to server.js such as account_management_app.js, schedule_planner_app.js  (Example will be provided)
- [ ] create route files in routes folder, create controller file in controller folder, create crud files in models folder
- [ ] use the express mini apps in server.js file using ``` app.use("account_management_app") ```

### PR DESCRIPTION
**Pull request by sontushkpaul**
## Source 

Closes #30 

### Summary of PR 

The PR will refactor some code from the server.js and make it as a module and implements the MVC design pattern (excluding View).

### Purpose 

The main goal of the PR is break the server into multiple mini app so that it is easy to maintain.  

## Review and Testing Steps

1. Pull the new branch 
2. Read the code (It is very important to understand why I wrote code in this way. Please ask question if you have any confusion or concerns because we should write code in the same structure in future. I have also added some TODO in the code for who will develop the mini app further)
3. Run the server and see if it works
4. Test the user registration routes and user login routes (can be found in routes folder)
5. Get the user list in the get user list routes 

If everything works then the code are fine otherwise let me know.

### Notes
The PR will not contain server configuration or database connection configuration files.

### Resources 
Want to learn about MVC design pattern in NODE.JS app?  Check out the video : https://www.youtube.com/watch?v=zW_tZR0Ir3Q

### CODE REVIEW

File: app/server/server.js [hunk 1]
```diff
@@ -1,96 +1,112 @@
...
+//     }
+//   } catch (err) {
+//     console.log(err)
+//   }
+// })
+
+
+
+
+// TODO: add a module for connecting to the database
```
Comment (jyaneshseeam) [inline app/server/server.js:90]: Looks good, I will create database.js for the database connection, then requiring it in the serverjs

**Review by jyaneshseeam — APPROVED** [review]
Everything looks good and structured, I will follow this code structure from now one!

**Comment by DemiAdisa** [comment]
Code looks well organized although, while I was trying out the users/register route, I got this error "HTTP/1.1 500 Internal Server Error".
This occurs when I try to register a new user

**Comment by sontushkpaul** [comment]
> Code looks well organized although, while I was trying out the users/register route, I got this error "HTTP/1.1 500 Internal Server Error". This occurs when I try to register a new user

@DemiAdisa  Fixed the bug . you can try again

**Comment by DemiAdisa** [comment]
> > Code looks well organized although, while I was trying out the users/register route, I got this error "HTTP/1.1 500 Internal Server Error". This occurs when I try to register a new user
> 
> @DemiAdisa Fixed the bug . you can try again

I am still getting that error unfortunately

**Comment by DemiAdisa** [comment]
> > > Code looks well organized although, while I was trying out the users/register route, I got this error "HTTP/1.1 500 Internal Server Error". This occurs when I try to register a new user
> 
> > 
> 
> > @DemiAdisa Fixed the bug . you can try again
> 
> 
> 
> I am still getting that error unfortunately
> 
> 

Error sorted out, can be merged now

**Review by DemiAdisa — APPROVED** [review]
All is good

**Review by ja-chisholm — APPROVED** [review]
Looks good!

**Pull request merged**